README.md

```diff
@@ -177,44 +177,64 @@ Example of a partial CR:
 * value of `ServiceBinding`'s global `bindAs` element, which applies to all binding data.
 * value of the `bindAs` attribute in each of the `dataMappings` elements inside `ServiceBinding`.
 
-#### Injecting data
-
-The key `SERVICE_BINDINGS` acts as a global map of the service bindings and **MUST** always be injected into the environment.  It contains a JSON payload with `bindingKeys` key name containing a list of all available binding information available. Each item of the `bindingKeys` list includes an object containing `name`, `bindAs` and an optional `mountPath` (if it is bound as a volume).
-
-Example:
-
-```json
-SERVICE_BINDINGS = {
-  "bindingKeys": [
-    {
-      "name": "KAFKA_USERNAME",
-      "bindAs": "envVar"
-    },
-    {
-      "name": "KAFKA_PASSWORD",
-      "bindAs": "volume",
-      "mountPath": "/platform/bindings/secret/"
-    }
-  ]
-}
-```
+If a particular binding key is made available as an environment variable an entry **MUST** be made at the following file:
 
-In the example above, the application **MAY** query the environment variable `SERVICE_BINDINGS`, walk its JSON payload and learn that `KAFKA_USERNAME` is available as an environment variable, and that `KAFKA_PASSWORD` is available as a mounted file inside the directory `/platform/bindings/secret/`.
+```
+<mountPathPrefix>/<service-id>/metadata/envVars
+```
 
-#### Mounting data
-Implementations of this specification must bind the following data into the consuming application container:
+Where:
+* `envVars` is a file where each line contains an environment variable name, representing one of the binding properties for this service.  
 
+For example, `platform/bindings/mongodb/metadata/envVars` could have:
 ```
-<mountPathPrefix>/bindings/secret/<persisted_secret>
+MONGODB_HOST
+MONGODB_PORT
 ```
 
-Where:
-* `<mountPathPrefix>` defaults to `platform` if not specified in the `ServiceBinding` CR via the `mountPathPrefix` element.
-* `<persisted_secret>` represents a set of files where the filename is a Secret key and the file contents is the corresponding value of that key.
+#### Guidelines for intermediate binding representation
+
+This specification does not mandate a particular methodology for implementations to process `ServiceBinding` CRs and mount the resulting data as per the requirements set in the previous sections, which allows for different frameworks to have flexibility in exactly how they fulfill the binding request.
+
+However, it is **RECOMMENDED** that implementations expose their intermediate binding representation (i.e. the model containing the binding data prior to mount) in the following way:
+
+* a single Secret, whose name matches the corresponding `ServiceBinding` CR's `metadata.name` and resides in the same namespace.  Each item inside this Secret is in the form of either:
+  * `<service-id>.secret.<binding-name>: <binding-value>`, representing a single binding item. Example: 
```

Comment:
@arthurdm what are your thoughts on including `type` info from [Service Binding Schema](https://github.com/application-stacks/service-binding-specification#service-binding-schema)?
e.g. `<service-id>.<type>.secret.<binding-name>: <binding-value>`

So app developers can easily query what data is available for particular type e.g. `db`, `kafka`

Alternatively, it is possible to just rely on `<service-id>.metadata.<metadata-name>: <metadata-value>` for convey that information e.g. `<service-id>.metadata.type: kafka` but it's just more work for a dev to find what `service-id` is for a particular `type`

Reply:
this definitely solves this [problem](https://github.com/application-stacks/service-binding-specification/pull/41#discussion_r430713400) , but add complexity for the application developer to find what `service-id` is for a particular type.

Reply:
that's an interesting question.  At the moment `type` is part of the binding schema, so we're advocating for it to be inside the `secret` path.  But it does feel more natural to be in `metadata`.  Something we'll need to solve.  I would vote for keeping the type out of the path, because I am not sure there's a reasonable default.

Reply:
The CNB Bindings spec defines two required bits of metadata `kind` and `provider`. The kind is particularly important at runtime so an application can consume a service by kind instead of by name.

Reply:
yeah - we'll probably need another future PR for the recommended set of metadata.